packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts

```diff
@@ -0,0 +1,215 @@
+              namedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#type'),
+              namedNode('http://www.w3.org/ns/ldp#Resource')));
+
+          const bodyStream = serialize.serialize(streamify(quads), { contentType: 'text/turtle' });
+          const body = await streamToBuffer(bodyStream);
+
+          return new Response(body, { status: 200, headers: { 'Content-Type': 'text/turtle' } });
+
+        } else {
+
+          // ComponentPodStore is requesting the metadata of a pod
+
+          const componentMock = components.find((component) => url === `${podUri}${component.filename}`);
+
+          if (componentMock) {
+
+            const quads: Quad[] = quadTransformer.toQuads([ componentMock.component ]);
+            const bodyStream = serialize.serialize(streamify(quads), { contentType: 'text/turtle' });
+            const body = await streamToBuffer(bodyStream);
+
+            return new Response(body, { status: 200, headers: { 'Content-Type': 'text/turtle' } });
+
+          }
+
+        }
+
+      }
+
+      return new Response('', { status: 501 });
+
+    });
+
+  };
+
+  beforeEach(() => {
+
+    jest.resetAllMocks();
+
+  });
+
+  it('should be correctly instantiated', () => {
+
+    expect(componentPodStore).toBeTruthy();
+
+  });
+
+  describe('all()', () => {
+
+    it('works if there are no components', async () => {
+
+      mockComponents([]);
+
+      await expect(componentPodStore.all()).resolves.toEqual([]);
+
+    });
+
+    it('works if there is 1 component', async () => {
+
+      const componentInfo = {
+        component: {
+          uri: 'https://components.semcom.digita.ai/components/test.component.js',
+          label: 'test component',
+          description: 'abc',
+          author: '',
+          tag: 'test',
+          version: '0.123',
+          latest: true,
+          shapes: [ 'http://digita.ai/voc/test#test' ],
+        },
+        filename: 'test1.ttl',
+      };
+
+      mockComponents([ componentInfo ]);
+
+      await expect(componentPodStore.all()).resolves.toEqual([ componentInfo.component ]);
+
+    });
+
+    it('works if there are multiple components', async () => {
+
+      const componentInfos = [ {
+        component: {
+          uri: 'https://components.semcom.digita.ai/components/test.component.js',
+          label: 'test component',
+          description: 'abc',
+          author: '',
+          tag: 'test',
+          version: '0.123',
+          latest: true,
+          shapes: [ 'http://digita.ai/voc/test#test' ],
+        },
+        filename: 'test1.ttl',
+      },
+      {
+        component: {
+          uri: 'https://components.semcom.digita.ai/components/test2.component.js',
+          label: 'test2',
+          description: 'defg',
+          author: '',
+          tag: 'test2',
+          version: '0.123',
+          latest: true,
+          shapes: [ 'http://digita.ai/voc/test#test2' ],
+        },
+        filename: 'test2.ttl',
+      } ];
```

Comment:
If you use these components multiple times, it's better to set them in a constant at the top of the file, and then set them where you need them.

Would become: `const componentInfos = [ component1, component2 ]` for example

Reply:
If you need to make changes to the components, you can also declare them globally with let, define them in a beforeEach, and then change only what needs to be changed in the individual test.